numpyro/util.py

```diff
@@ -174,9 +174,9 @@ def fori_collect(lower, upper, body_fun, init_val, transform=identity,
         be any Python collection type containing `np.ndarray` objects.
     :param transform: a callable to post-process the values returned by `body_fn`.
     :param progbar: whether to post progress bar updates.
-    :param bool return_init_state: If `True`, the state at iteration `lower-1`,
-        where the collection begins, is also returned. This has the same type
-        as `init_val`.
+    :param bool return_last_val: If `True`, the last value is also returned.
+        This has the same type as `init_val`.
+    :param int collect_size: Size of the returned collection.
```

Comment:
Could you add that the collection size will be `upper-lower` if this is not specified, and also that if this is larger than `upper-lower` only the top `upper-lower` entries will be non-zero. I think a more general utility should allow for a collection argument (instead of size) that will be used to fill up the entries, but since we are only using this specifically for mcmc, we can keep it as is for now.